controllers/automation/policyautomation_controller.go

```diff
@@ -89,6 +92,150 @@ func (r *PolicyAutomationReconciler) setOwnerReferences(
 	return nil
 }
 
+// getTargetListMap will convert slice targetList to map for search efficiency
+func getTargetListMap(targetList []string) map[string]bool {
+	targetListMap := map[string]bool{}
+	for _, target := range targetList {
+		targetListMap[target] = true
+	}
+
+	return targetListMap
+}
+
+// getClusterName will get the hub cluster name information
+func getClusterName() string {
+	// Get a config to talk to the apiserver
+	cfg := config.GetConfigOrDie()
+	hostStr := cfg.Host
+	hubCluster := hostStr
+
+	if len(hostStr) > 0 {
+		// remove possible https head
+		hostStr := strings.ReplaceAll(cfg.Host, "https://", "")
+		// remove possible http head
+		hostStr = strings.ReplaceAll(hostStr, "http://", "")
+		// remove possible port number
+		hostStr = strings.SplitN(hostStr, ":", 2)[0]
+		hubCluster = hostStr
+		// convert host name to URL if it is IP address
+		if net.ParseIP(hostStr) != nil {
+			hostAddr, err := net.LookupAddr(hostStr)
+			if err == nil {
+				hubCluster = hostAddr[0]
+			}
```

Comment:
After looking at the docs, it's unclear when an error is returned but it does seem that an error can be returned even when some results are returned, so I think this is safer:
```suggestion
			names, _ := net.LookupAddr(hostStr)
			if len(names) > 1 {
				hubCluster = names[0]
			}
```